src/views/ClosePurchaseOrderModal.vue

```diff
@@ -0,0 +1,177 @@
+      <ion-buttons slot="end" @click="selectAllItems">
+        <ion-button color="primary">{{ $t("Select all") }}</ion-button>
+      </ion-buttons>
+    </ion-toolbar>
+  </ion-header>
+
+  <ion-content>
+    <ion-item lines="none">
+      <ion-list-header>{{ $t("To close the purchase order, select all.") }}</ion-list-header>
+    </ion-item>
+    <ion-list v-for="(item, index) in order.items" :key="index">
+      <ion-item>
+        <ion-thumbnail slot="start">
+          <ShopifyImg size="small" :src="getProduct(item.productId).mainImageUrl" />
+        </ion-thumbnail>
+        <ion-label>
+          <h2>{{ productHelpers.getProductIdentificationValue(productIdentificationPref.primaryId, getProduct(item.productId)) }}</h2>
+          <p>{{ productHelpers.getProductIdentificationValue(productIdentificationPref.secondaryId, getProduct(item.productId)) }}</p>
+        </ion-label>
+        <ion-buttons>
+          <ion-badge v-if="item.orderItemStatusId === 'ITEM_COMPLETED'" slot="end">{{ $t("Completed") }}</ion-badge>
+          <ion-badge v-else-if="item.orderItemStatusId === 'ITEM_REJECTED'" color="danger" slot="end">{{ $t("Rejected") }}</ion-badge>
+          <ion-checkbox v-else slot="end" v-model="item.isChecked" />
+        </ion-buttons>
+      </ion-item>
+    </ion-list>
+  </ion-content>
+
+  <ion-fab vertical="bottom" horizontal="end" slot="fixed">
+    <ion-fab-button :disabled="!hasPermission(Actions.APP_SHIPMENT_UPDATE) || !isEligibleToClosePOItems()" @click="confirmSave">
+      <ion-icon :icon="saveOutline" />
+    </ion-fab-button>
+  </ion-fab>
+</template>
+
+<script lang="ts">
+import {
+  alertController,
+  IonButton,
+  IonButtons,
+  IonBadge,
+  IonCheckbox,
+  IonContent,
+  IonFab,
+  IonFabButton,
+  IonHeader,
+  IonIcon,
+  IonItem,
+  IonLabel,
+  IonList,
+  IonListHeader,
+  IonTitle,
+  IonToolbar,
+  IonThumbnail,
+  modalController
+} from '@ionic/vue';
+import { Actions, hasPermission } from '@/authorization'
+import { closeOutline, checkmarkCircle, arrowBackOutline, saveOutline } from 'ionicons/icons';
+import { defineComponent } from 'vue';
+import { mapGetters } from 'vuex'
+import { OrderService } from "@/services/OrderService";
+import { productHelpers, showToast } from '@/utils';
+import { ShopifyImg } from '@hotwax/dxp-components';
+import { translate } from '@/i18n'
+
+export default defineComponent({
+  name: "ClosePurchaseOrder",
+  components: {
+    IonBadge,
+    IonButton,
+    IonButtons,
+    IonCheckbox,
+    IonContent,
+    IonFab,
+    IonFabButton,
+    IonHeader,
+    IonIcon,
+    IonItem,
+    IonLabel,
+    IonList,
+    IonListHeader,
+    IonTitle,
+    IonThumbnail,
+    IonToolbar,
+    ShopifyImg
+  },
+  computed: {
+    ...mapGetters({
+      getProduct: 'product/getProduct',
+      order: 'order/getCurrent',
+      productIdentificationPref: 'user/getProductIdentificationPref'
+    })
+  },
+  props: ['createShipment', 'isEligibileForCreatingShipment'],
```

Comment:
Instead of passing functions as props, we should use emitter as passing function props is an anti-pattern in Vue.